RFS/src/main/java/org/opensearch/migrations/bulkload/common/OpenSearchClientFactory.java

```diff
@@ -66,7 +66,7 @@ public OpenSearchClient determineVersionAndCreate(RestClient restClient, FailedR
     private Class<? extends OpenSearchClient> getOpenSearchClientClass(Version version) {
         if (VersionMatchers.isOS_1_X.or(VersionMatchers.isOS_2_X).or(VersionMatchers.isES_7_X).test(version)) {
             return OpenSearchClient_OS_2_11.class;
-        } else if (VersionMatchers.isES_6_X.test(version)) {
+        } else if (VersionMatchers.isES_6_X.or(VersionMatchers.isES_5_X).test(version)) {
```

Comment:
I'm confused - are we adding support for ES5 indicies in ES6, or support for ES5, in this PR?

Reply:
Good eye - I'm doing a little ES5 work here that makes the testing easier.

For RFS's purposes the lucene version of the segments in the snapshot is the key difference.  By adding ES5 in the EndToEnd tests we get the testing coverage rather than building a new test that creates an ES5 cluster, creates indices and documents, upgrades to ES6, and then migrates.

Note; I've updated https://opensearch.atlassian.net/browse/MIGRATIONS-2276 's AC to include adding lucene6 when handing ES5 clusters because we will have the same issue with ES2 -> ES5 upgraded clusters.

Reply:
Per convo, this change means that we will unofficially be able to handle ES 5 source domains for RFS.  That is, if someone points RFS at such a domain, it may just work even though it's not in our compatibility matrix.  Metadata migrations will fail due to lack of a transform path.  This all seems fine, as we're adding full ES 5 support next anyways.